server/service/src/document/patient/query.rs

```diff
@@ -0,0 +1,142 @@
+use chrono::NaiveDate;
+use repository::{
+    DateFilter, EqualFilter, Gender, NameFilter, NameRepository, NameRow, NameSort, NameSortField,
+    NameType, PaginationOption, RepositoryError, SimpleStringFilter, Sort,
+};
+
+use crate::{get_default_pagination_unlimited, service_provider::ServiceContext};
+
+#[derive(Clone, Default)]
+pub struct PatientFilter {
```

Comment:
Shouldn't we just use name filter and map to it from patient graphql ?

Reply:
Could do, will leave it for now (still pondering about having a separate patient table...)

Reply:
sure, but we've moved away from defining types like this in service, this mapping is now done in graphql